api/src/main/java/org/openmrs/PatientProgram.java

```diff
@@ -398,4 +404,86 @@ public void setLocation(Location location) {
 		Collections.sort(sortedStates);
 		return sortedStates;
 	}
+
+        @Override
+        public Set<PatientProgramAttribute> getAttributes() {
+            return attributes;
+        }
+
+        @Override
+        public Collection<PatientProgramAttribute> getActiveAttributes() {
+            ArrayList<PatientProgramAttribute> ret = new ArrayList<>();
+
+            if (this.getAttributes() != null) {
+                for (PatientProgramAttribute attr : this.getAttributes()) {
+                    if (!attr.isVoided()) {
+                        ret.add(attr);
+                    }
+                }
+            }
+
+            return ret;
+        }
+
+        @Override
+        public List<PatientProgramAttribute> getActiveAttributes(CustomValueDescriptor ofType) {
+            ArrayList<PatientProgramAttribute> ret = new ArrayList<>();
+
+            if (this.getAttributes() != null) {
+                for (PatientProgramAttribute attr : this.getAttributes()) {
+                    if (attr.getAttributeType().equals(ofType) && !attr.isVoided()) {
+                        ret.add(attr);
+                    }
+                }
+            }
+
+            return ret;
+        }
+
+        @Override
+        public void addAttribute(PatientProgramAttribute attribute) {
+            if (this.getAttributes() == null) {
+                this.setAttributes(new LinkedHashSet());
+            }
+
+            this.getAttributes().add(attribute);
+            attribute.setOwner(this);
+        }
+
+        public void setAttributes(Set<PatientProgramAttribute> attributes) {
+            this.attributes = attributes;
+        }
+
+        public void setAttribute(PatientProgramAttribute attribute) {
+            if (this.getAttributes() == null) {
+                this.addAttribute(attribute);
+            } else {
+                if (this.getActiveAttributes(attribute.getAttributeType()).size() == 1) {
+                    PatientProgramAttribute patientProgramAttribute = this.getActiveAttributes(attribute.getAttributeType()).get(0);
+                    if (!patientProgramAttribute .getValue().equals(attribute.getValue())) {
```

Comment:
Did you intentionally leave the space before .getValue()?

Reply:
That's directly from the bahmni source

Reply:
Can you point me exactly where that is?